pkg/kubecfg/update.go

```diff
@@ -512,7 +525,23 @@ func eligibleForGc(obj metav1.Object, gcTag string) bool {
 		strategy = GcStrategyAuto
 	}
 
-	// [gctag-migration]: Check *label* == tag instead in phase2
-	return a[AnnotationGcTag] == gcTag &&
-		strategy == GcStrategyAuto
+	if strategy == GcStrategyAuto && a[AnnotationGcTag] != "" {
```

Comment:
I think this is a good opportunity to make this new feature use the `kubecfg.ksonnet.io/garbage-collect-tag` *label* and not the annotation.

I've been a bit "slow" (euphemism) to implement the migration from annotation to labels, but eventually this migration will happen.

It would be nice if we don't have to worry about people who checked in jsonnet in the VCSs with  `annotations: {'kubecfg.ksonnet.io/garbage-collect-tag': 'foo'}` without the corresponding label.

Reply:
Hello @mkmik, Thank you for reviewing! I've changed the code such that the label is preferred over the annotation.

Playing around, I observed problems with metric-server's podmetrics which get labels of the pods they belong to, but no owner reference.

Reply:
> Playing around, I observed problems with metric-server's podmetrics which get labels of the pods they belong to, but no owner reference.

I don't think I understood, could you please describe the problem a bit more in details?

(sorry I was a bit busy; I should be a bit more responsive now)

Reply:
wow. So basically we need labels in order to search but we need also use the annotation to further filter the resultset